virtio-devices/src/transport/pci_device.rs

```diff
@@ -847,19 +839,31 @@ impl PciDevice for VirtioPciDevice {
         &mut self,
         allocator: &Arc<Mutex<SystemAllocator>>,
         mmio_allocator: &mut AddressAllocator,
+        resources: Option<Vec<Resource>>,
     ) -> std::result::Result<Vec<(GuestAddress, GuestUsize, PciBarRegionType)>, PciDeviceError>
     {
         let mut ranges = Vec::new();
         let device_clone = self.device.clone();
         let device = device_clone.lock().unwrap();
+        let settings_bar_addr = if let Some(resources) = &resources {
+            if resources.is_empty() {
+                return Err(PciDeviceError::MissingResource);
+            }
```

Comment:
As we are expecting exact one entry/bar for the VirtioPciDevice, may worth reporting a warning for case that `resources.len() > 1`?

Reply:
We might have 2. Very unlikely to happen but this is related to the way virtio-fs works.
I can constraint it but eventually I think it'd be nice to store all kind of resources (interrupts, BAR, ranges), meaning we wouldn't have only 1 or 2 resources.